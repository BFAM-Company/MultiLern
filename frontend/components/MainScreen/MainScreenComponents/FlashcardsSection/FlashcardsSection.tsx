import React from 'react';
import { Animated, Image, StyleSheet, Text, TouchableOpacity, View} from 'react-native';

function FlashcardsSection({pageSwitcher}: any) {
  return (
    <Animated.View
        style={styles.mainContainer}
    >
        <View style={styles.TopHeader}>
            <Text style={styles.title}>Fiszki</Text>
        </View>
        <TouchableOpacity
            style={styles.container}
            testID='myPageSwitcher'
            onPress={ () => pageSwitcher('FlashcardsList', {range: 'my'})}
        >
            <Image
                style={styles.image}
                source={require('../../../../assets/flashcards-icon2.png')}
            />
            <Text style={styles.buttonTitle}>Twoje Fiszki</Text>
            <TouchableOpacity onPress={ () => pageSwitcher('FlashcardsList', {range: 'all'})}><Text testID='seeAllFiches' style={styles.buttonDescription}>Zobacz wszystkie</Text></TouchableOpacity>
            <TouchableOpacity
                style={styles.button}
                onPress={ () => pageSwitcher('FlashcardsList', {range: 'my'})}
            >
                <Text style={styles.buttonText}>Dalej</Text>
            </TouchableOpacity>
        </TouchableOpacity>
        <TouchableOpacity
            style={styles.container}
            onPress={ () => pageSwitcher('NewFlashcard')}
        >
            <Image
                style={styles.image}
                source={require('../../../../assets/flashcards-icon.png')}
                resizeMode='contain'
                
            />
             <Text style={styles.buttonTitle}>Nowe Fiszki</Text>
             <Text style={styles.buttonDescription}>Utwórz nowy set</Text>
             <TouchableOpacity
                style={styles.button}
                onPress={ () => pageSwitcher('NewFlashcard')}
            >
                <Text style={styles.buttonText}>Utwórz +</Text>
            </TouchableOpacity>
        </TouchableOpacity>
    </Animated.View>
  );
}

const styles = StyleSheet.create({
    mainContainer:{
        width:'100%',
        display:'flex',
        flexDirection:'row',
        flexWrap:'wrap',
        justifyContent:'space-around',
        alignItems:'center',
    },
    TopHeader:{
        width:'100%',
        display:'flex',
        alignItems:'flex-start',
        justifyContent:'center',
        padding:20,
    },
    title:{
        fontSize:24,
        fontWeight:'900',
        color:'rgb(33,33,43)',
    },
    container:{
        width:180,
        height:300,
        display:'flex',
        alignItems:'center',
        justifyContent:'space-between',
        padding:10,
        borderRadius:20,
        backgroundColor:'#fff',
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 0,
        },
        shadowOpacity: 0.58,
        shadowRadius: 16.00,

        elevation: 20,
    },
    image:{
        width:170,
        height:170,
        borderRadius:20,
        backgroundColor:'#F5F5F5',
    },
    buttonTitle:{
        color:'rgb(33,33,43)',
        fontSize:18,
        fontWeight:'900',
        padding:5,
    },
    button:{
        width:'90%',
        height:30,
        display:'flex',
        alignItems:'center',
        justifyContent:'center',
        backgroundColor:'rgb(33,33,43)',
        borderRadius:20,
    },
    buttonText:{
        color:'#fff',
        fontWeight:'500',
        fontSize:16,
    },
    buttonDescription:{
        color:'#c085ff',
        fontSize:16,
        fontWeight:'500',
        padding:5,
    }
})

export default FlashcardsSection;