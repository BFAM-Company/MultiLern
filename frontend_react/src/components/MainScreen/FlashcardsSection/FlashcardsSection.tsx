import React from 'react';
import './FlashcardsSection.css';
import { useNavigate } from 'react-router-dom';

function FlashcardsSection() {
    const navigate = useNavigate()
  return (
    <div className="mainContainer">
      <div className="topHeader">
        <h2 className="title">Fiszki</h2>
      </div>
      <div 
        className="container" 
        onClick={() => navigate('/flashcardsList?range=my')}
      >
        <img 
          className="image" 
          src="/assets/flashcards-icon2.png" 
          alt="Flashcards Icon" 
        />
        <h3 className="buttonTitle">Twoje Fiszki</h3>
        <button 
          className="buttonDescription" 
          onClick={(e) => {
            e.stopPropagation();
            navigate('/flashcardsList?range=all');
          }}
        >
          Zobacz wszystkie
        </button>
        <button 
          className="button" 
          onClick={(e) => {
            e.stopPropagation();
            navigate('/flashcardsList?range=my');
          }}
        >
          Dalej
        </button>
      </div>
      <div 
        className="container" 
        onClick={() => navigate('/newFlashcard')}
      >
        <img 
          className="image" 
          src="/assets/flashcards-icon.png" 
          alt="New Flashcards Icon" 
        />
        <h3 className="buttonTitle">Nowe Fiszki</h3>
        <p className="buttonDescription">Utwórz nowy set</p>
        <button 
          className="button" 
          onClick={(e) => {
            e.stopPropagation();
            navigate('/newFlashcard');
          }}
        >
          Utwórz +
        </button>
      </div>
    </div>
  );
}

export default FlashcardsSection;
