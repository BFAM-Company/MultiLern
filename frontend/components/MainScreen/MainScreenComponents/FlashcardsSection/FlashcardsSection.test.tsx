import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react-native';
import FlashcardsSection from './FlashcardsSection';

describe('FlashcardsSection', () => {
  afterEach(cleanup);
    it('renders correctly', () => {
        const mockPageSwitcher = jest.fn();
        const { toJSON } = render(<FlashcardsSection pageSwitcher={mockPageSwitcher} />);
        expect(toJSON()).toMatchSnapshot();
    });

    it('navigates to FlashcardsList with "my" range when "Twoje Fiszki" button is pressed', () => {
        const mockPageSwitcher = jest.fn();
        const { getByText } = render(<FlashcardsSection pageSwitcher={mockPageSwitcher} />);
        fireEvent.press(getByText('Dalej'));

        expect(mockPageSwitcher).toHaveBeenCalledWith('FlashcardsList', { range: 'my' });
    });

    it('navigates to FlashcardsList with "my" range when card is pressed', () => {
        const mockPageSwitcher = jest.fn();
        const { getByTestId } = render(<FlashcardsSection pageSwitcher={mockPageSwitcher} />);
        fireEvent.press(getByTestId('myPageSwitcher'));

        expect(mockPageSwitcher).toHaveBeenCalledWith('FlashcardsList', { range: 'my' });
    });

    it('navigates to FlashcardsList with "all" range when "Zobacz wszystkie" button is pressed', () => {
        const mockPageSwitcher = jest.fn();
        const { getByText } = render(<FlashcardsSection pageSwitcher={mockPageSwitcher} />);
        fireEvent.press(getByText('Zobacz wszystkie'));

        expect(mockPageSwitcher).toHaveBeenCalledWith('FlashcardsList', { range: 'all' });
    });

    it('navigates to NewFlashcard when "Nowe Fiszki" button is pressed', () => {
        const mockPageSwitcher = jest.fn();
        const { getByText } = render(<FlashcardsSection pageSwitcher={mockPageSwitcher} />);
        fireEvent.press(getByText('UtwÃ³rz +'));

        expect(mockPageSwitcher).toHaveBeenCalledWith('NewFlashcard');
    });

    it('test pageSwitcher Nowe Fiszki', () => {
        const mockPageSwitcher = jest.fn();
        const { getByText } = render(<FlashcardsSection pageSwitcher={mockPageSwitcher} />);
        fireEvent.press(getByText('Nowe Fiszki'));

        expect(mockPageSwitcher).toHaveBeenCalledWith('NewFlashcard');
    });
});
