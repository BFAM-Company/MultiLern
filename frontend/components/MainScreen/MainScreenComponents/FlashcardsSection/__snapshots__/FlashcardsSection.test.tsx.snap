// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashcardsSection renders correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "space-around",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "display": "flex",
        "justifyContent": "center",
        "padding": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "rgb(33,33,43)",
          "fontSize": 24,
          "fontWeight": "900",
        }
      }
    >
      Fiszki
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 20,
        "display": "flex",
        "elevation": 20,
        "height": 300,
        "justifyContent": "space-between",
        "opacity": 1,
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.58,
        "shadowRadius": 16,
        "width": 180,
      }
    }
    testID="myPageSwitcher"
  >
    <Image
      source={1}
      style={
        {
          "backgroundColor": "#F5F5F5",
          "borderRadius": 20,
          "height": 170,
          "width": 170,
        }
      }
    />
    <Text
      style={
        {
          "color": "rgb(33,33,43)",
          "fontSize": 18,
          "fontWeight": "900",
          "padding": 5,
        }
      }
    >
      Twoje Fiszki
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#c085ff",
            "fontSize": 16,
            "fontWeight": "500",
            "padding": 5,
          }
        }
      >
        Zobacz wszystkie
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgb(33,33,43)",
          "borderRadius": 20,
          "display": "flex",
          "height": 30,
          "justifyContent": "center",
          "opacity": 1,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        Dalej
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 20,
        "display": "flex",
        "elevation": 20,
        "height": 300,
        "justifyContent": "space-between",
        "opacity": 1,
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.58,
        "shadowRadius": 16,
        "width": 180,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        {
          "backgroundColor": "#F5F5F5",
          "borderRadius": 20,
          "height": 170,
          "width": 170,
        }
      }
    />
    <Text
      style={
        {
          "color": "rgb(33,33,43)",
          "fontSize": 18,
          "fontWeight": "900",
          "padding": 5,
        }
      }
    >
      Nowe Fiszki
    </Text>
    <Text
      style={
        {
          "color": "#c085ff",
          "fontSize": 16,
          "fontWeight": "500",
          "padding": 5,
        }
      }
    >
      Utwórz nowy set
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgb(33,33,43)",
          "borderRadius": 20,
          "display": "flex",
          "height": 30,
          "justifyContent": "center",
          "opacity": 1,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        Utwórz +
      </Text>
    </View>
  </View>
</View>
`;
